### DISCUSSION TITLE
What's the best practice for extend a class by mapping

### DISCUSSION THREAD

**Discussion by PaleNeutron** [sqlalchemy/sqlalchemy#12120] [Usage Questions] [answered]
Based on this section https://docs.sqlalchemy.org/en/20/orm/inheritance.html#single-table-inheritance .

If I create a `Engineer` class first and used in a lot of apis. Then my system growth, I have to support `Manager`. At this point, all my api's need to add `.where(Engineer.type == 'engineer')`, and very likely to miss some api and cause bugs. `select(Engineer)` will also return `Manager` objects which is very confusing.

Is there a way to auto assign the filter to any where I use `Engineer` just like I have create a view named `engineer` with the filter?

**Comment by zzzeek** [comment, marked as answer]
> select(Engineer) will also return Manager objects which is very confusing.

in those docs, `Manager` is not a subclass of `Engineer`, it's a sibling.  so `select(Engineer)` only returns `Engineer` objects, and nothing needs to change.

OTOH, if you were adding to your code a subclass of `Engineer`, like `MechanicalEngineer`, then your existing queries that query for `Engineer` we would assume are fine for the most part since a `MechanicalEngineer` is an `Engineer`.
